server.py

```diff
@@ -156,7 +156,9 @@ def update_recommendations(id):
 def delete_recommendations(id):
     if get_recommendations(id).status_code == 200:
         conn.execute("DELETE FROM recommendations WHERE id=%d" % id)
-    return '', HTTP_204_NO_CONTENT
+        return '', HTTP_204_NO_CONTENT
```

Comment:
I believe it was that DELETE is idempotent, so it should not return 404 on subsequent attempts. It should return 204.